api/python/t4/packages.py

```diff
@@ -647,9 +647,14 @@ def _fix_sha256(self):
         if not entries:
             return
 
-        physical_keys = (entry.physical_keys[0] for entry in entries)
-        total_size = sum(entry.size for entry in entries)
-        results = calculate_sha256(physical_keys, total_size)
+        physical_keys = []
+        sizes = []
+        for entry in entries:
+            physical_keys.append(entry.physical_keys[0])
+            sizes.append(entry.size)
+
+        total_size = sum(sizes)
+        results = calculate_sha256(physical_keys, sizes)
```

Comment:
I _think_ a better API would be a list of tuples (path, size) - though I don't have a strong opinion.

But if you do leave it as is, then add an assert to `calculate_sha256` to make sure they have the same length.

Reply:
Did the latter.